ci/update-intellij/src/main/scala/org.virtuslab.ideprobe/BumpIntelliJ.scala

```diff
@@ -0,0 +1,100 @@
+import java.io.PrintWriter
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+
+import scala.io.Source
+import scala.jdk.CollectionConverters._
+
+import org.jsoup.Jsoup
+
+object BumpIntelliJ extends App {
+
+  private val referenceConfFile = new File("core/driver/sources/src/main/resources/reference.conf")
+  private val ideProbeConfigTestFile =
+    new File("core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/IdeProbeConfigTest.scala")
+
+  private val filesToUpdate = Seq(
+    referenceConfFile,
+    ideProbeConfigTestFile
+  )
+
+  private val releaseRegex = """20\d\d.\d.?\d?"""
+
+  private val intellijReleaseFromReferenceConf = getValueFromReferenceConf("release = \"")
+  private val intellijBuildNumberFromReferenceConf = getValueFromReferenceConf("build = \"")
+
+  private val officialIntellijReleaseLinks: List[String] =
+    Jsoup
+      .connect("https://www.jetbrains.com/intellij-repository/releases/")
+      .get()
+      .select(
+        "a[href^=https://www.jetbrains.com/intellij-repository/releases/com/jetbrains/intellij/idea/ideaIC/][href$=.pom]"
+      )
+      .eachAttr("href")
+      .asScala
+      .toList
+
+  private val officialIntellijReleases =
+    officialIntellijReleaseLinks
+      .map(link => link.substring(link.indexOf("ideaIC-"), link.indexOf(".pom")).replace("ideaIC-", ""))
+      .filter(_.matches(releaseRegex))
+
+  private val newestStableRelease = officialIntellijReleases.max
+
+  private lazy val newestStableBuildNumber = {
+    val newestStableReleasePomLink = officialIntellijReleaseLinks.find(_.contains(newestStableRelease)).get
+    def replaceIdeaICPomFileNameWithBuildTxt(ideaICpomFileName: String): String =
+      ideaICpomFileName
+        .replace("ideaIC-", "BUILD-")
+        .replace(".pom", ".txt")
+        .replace("ideaIC", "BUILD")
+
+    val newestStableBuildNumberFileURL = new URL(replaceIdeaICPomFileNameWithBuildTxt(newestStableReleasePomLink))
+    Source.fromURL(newestStableBuildNumberFileURL).mkString
+  }
+
+  if (newestStableRelease > intellijReleaseFromReferenceConf) filesToUpdate.foreach(bumpIntellijVersion)
+
+  private def bumpIntellijVersion(oldFile: File): Unit = {
+    val newFile = new File("/tmp/bump_version.txt")
+    val w = new PrintWriter(newFile)
+    Source
+      .fromFile(oldFile)
+      .getLines()
+      .map { line =>
+        if (line.contains(intellijReleaseFromReferenceConf))
+          line.replace(intellijReleaseFromReferenceConf, newestStableRelease)
+        else line
+      }
+      .map { line =>
+        if (line.contains(intellijBuildNumberFromReferenceConf))
+          line.replace(intellijBuildNumberFromReferenceConf, newestStableBuildNumber)
+        else line
+      }
+      .foreach(x => w.println(x))
+    w.close()
+    Files.move(newFile.toPath, oldFile.toPath, REPLACE_EXISTING)
+  }
+
+  private def getValueFromReferenceConf(charsUntilValue: String): String = Source
```

Comment:
Maybe just read the file with Typesafe Config library? 🤔

Reply:
I would keep the logic it as it is for one reason - my current solution provides the same extracting logic both in `ci/update-intellij/src/main/scala/org.virtuslab.ideprobe/BumpIntelliJ.scala` and in `.github/workflows/update-intellij-version.yml`. We have:
- `private val intellijReleaseFromReferenceConf = getValueFromReferenceConf("release = \"")` in scala code
- `RELEASE_LINE=$(grep 'release = \"' core/driver/sources/src/main/resources/reference.conf)` in the script
which means - in both cases we are searching for `release = \"`.
If we change the scala logic to use Typesafe Config tricks (instead of my not-so-elegant logic) - then there will be possible scenarios, where someone changes the style in `reference.conf` - so that scala logic works fine, but the logic from script does **not** find the value...

If I should switch to use Typesafe Config library - then also the script should somehow get the same value that was extracted in scala code (some temporary file usage maybe?)

Reply:
Okay, makes sense... leaving up to Łukasz W for the final review